### PR TITLE
Allow making the plugin from multiple platforms

### LINKED ISSUE #6052: `bin/plugin make` fails on Windows
## Steps to Reproduce
- On a Windows operating system, run `bin/plugin` or `dart tool/plugin/bin/main.dart make`

## Result
The script uses OS specific calls and is not able to perform the necessary operations on Windows as it does on Linux.

## Version info
Info attached: [flutter_doctor.txt](https://github.com/flutter/flutter-intellij/files/8337139/flutter_doctor.txt)

### PR DESCRIPTION
**Pull request by AlexV525**
...especially for Windows Native (without WSL).

Resolves #6052 #6640 #6733

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: tool/plugin/lib/plugin.dart [hunk 1]
```diff
@@ -731,6 +705,7 @@ abstract class ProductCommand extends Command {
   Future<int> run() async {
     await _initGlobals();
     await _initSpecs();
+    _handleSymlinksOnWindows();
```
Comment (AlexV525) [inline tool/plugin/lib/plugin.dart:708]: This is the only necessary change for Windows currently.

Comment (stevemessick) [reply tool/plugin/lib/plugin.dart:708]: You can also create the local repo in such a way that symlinks work; see line 139 in CONTRIBUTING.md. But, now we don't have to remember to do that. Thanks!

**Review by stevemessick — APPROVED** [review]

File: tool/plugin/lib/plugin.dart (hunk 1 above)
Comment (AlexV525) [reply tool/plugin/lib/plugin.dart:708]: Oh sure that line can be removed too IMO.

**Comment by AlexV525** [comment]
I just experienced a full tour of the building and running the plugin on Windows. There are a couple of places that need to be updated in the contributing doc.

**Review by stevemessick — APPROVED** [review]
This is great, thanks!

**Pull request merged**